openwpm/task_manager.py

```diff
@@ -173,6 +165,18 @@ def __init__(
         self.callback_thread.name = "OpenWPM-completion_handler"
         self.callback_thread.start()
 
+    def __enter__(self):
```

Comment:
Why are these needed?

Reply:
These turn the TaskManager into a context manager, making it harder to forget to shut it down.